lib/chef/resource/powershell_package_source.rb

```diff
@@ -0,0 +1,163 @@
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/json_compat"
+
+class Chef
+  class Resource
+    class PowershellPackageSource < Chef::Resource
+      preview_resource true
+      resource_name "powershell_package_source"
+
+      description "Use the powershell_package_source resource to register a powershell package repository"
+      introduced "14.3"
+
+      property :source_name, String,
+               description: "The name of the package source",
+               name_property: true
+
+      property :url, String,
+               description: "The url to the package source",
+               required: true
+
+      property :trusted, [TrueClass, FalseClass],
+               description: "Whether or not to trust packages from this source",
+               default: false
+
+      property :provider_name, String,
+               equal_to: %w{ Programs msi NuGet msu PowerShellGet psl chocolatey },
+               validation_message: "The following providers are supported: 'Programs', 'msi', 'NuGet', 'msu', 'PowerShellGet', 'psl' or 'chocolatey'",
+               description: "The package management provider for the source. It supports the following providers: "\
+                            "'Programs', 'msi', 'NuGet', 'msu', 'PowerShellGet', 'psl' and 'chocolatey'.",
+               default: "NuGet"
+
+      property :publish_location, String,
+               description: "The url where modules will be published to for this source. Only valid if the provider is 'PowerShellGet'."
+
+      property :script_source_location, String,
+               description: "The url where scripts are located for this source. Only valid if the provider is 'PowerShellGet'."
+
+      property :script_publish_location, String,
+               description: "The location where scripts will be published to for this source. Only valid if the provider is 'PowerShellGet'."
+
+      load_current_value do
+        cmd = load_resource_state_script(name)
+        repo = powershell_out!(cmd)
+        status = Chef::JSONCompat.from_json(repo.stdout)
+        url status["url"].nil? ? "not_set" : status["url"]
+        trusted status["trusted"]
+        provider_name status["provider_name"]
+        publish_location status["publish_location"]
+        script_source_location status["script_source_location"]
+        script_publish_location status["script_publish_location"]
+      end
+
+      action :register do
```

Comment:
You'll want to include a description field for the actions as well. We'll be generating documentation off that sometime in the future

Reply:
Fixed.